_includes/author.html

```diff
@@ -1,35 +1,53 @@
+</div>
+<div class="author-info">
+  <a class="btn-link" href="{{ site.baseurl }}/authors/{{ post.author }}"
+    title="Articles by {{ author.display_name}}">{{ author.display_name }}</a>
+  {%- if categories.size > 0 -%} <span> on</span> {% for category in categories %}
+  <a class="btn-link" href="{{ site.baseurl }}/tags/{{ category | downcase | replace: ' ','-' }}"
+    title="Articles on {{ category | titleize | replace: '-',' ' }}">{{ category | titleize | replace: '-',' ' }}</a>
+  {%- unless forloop.last -%},{%- endunless -%}{% endfor %}
+  {%- endif -%}
+</div>
 {%- endif -%}
 
 {%- if authors -%}
-  {% for author in authors %}
-    {%- assign author_info = site.authors[author] -%}
-    {%- if author_info.gravatar -%}
-      <p class="author-info">
-        <img class="author-thumb" src="//2.gravatar.com/avatar/{{ author_info.gravatar }}?s=50" alt="{{ author_info.display_name }}" nopin="nopin" />
-    {%- else -%}
-        <img class="author-thumb" src="{{ site.baseurl }}/assets/images/profile.png" alt="Author's profile picture" nopin="nopin" />
-    {%- endif -%}
-        <a class="btn-link" href="{{ site.baseurl }}/authors/{{ author }}"
-        title="Articles by {{ author_info.display_name}}">{{ author_info.display_name }}</a>
-      </p>
-    {%- if authors.size == 2 -%}
-      {%- unless forloop.last -%}<p>and </p>{%- endunless -%}
-    {%- else -%}
-      {% if forloop.first == true %}
-        {%- unless forloop.last -%}<p>, </p>{%- endunless -%}
-      {%- else -%}
-        {%- unless forloop.last -%}<p>and </p>{%- endunless -%}
-      {%- endif -%}
-    {%- endif -%}
-  {% endfor %}
+{% for author in authors %}
+{%- assign author_info = site.authors[author] -%}
+{%- if author_info.gravatar -%}
+<div class="author-photo">
+  <img class="author-thumb" src="//2.gravatar.com/avatar/{{ author_info.gravatar }}?s=50"
+    alt="{{ author_info.display_name }}" nopin="nopin" />
+  {%- else -%}
+  <img class="author-thumb" src="{{ site.baseurl }}/assets/images/profile.png" alt="Author's profile picture"
+    nopin="nopin" />
+  {%- endif -%}
+</div>
+<div class="author-info">
+  <a class="btn-link" href="{{ site.baseurl }}/authors/{{ author }}"
+    title="Articles by {{ author_info.display_name}}">{{ author_info.display_name }}</a>
+  {%- if categories.size > 0 -%} <span> on</span> {% for category in categories %}
+  <a class="btn-link" href="{{ site.baseurl }}/tags/{{ category | downcase | replace: ' ','-' }}"
+    title="Articles on {{ category | titleize | replace: '-',' ' }}">{{ category | titleize | replace: '-',' ' }}</a>
+  {%- unless forloop.last -%},{%- endunless -%}{% endfor %}
+  {%- endif -%}
+</div>
+{%- if authors.size == 2 -%}
+{%- unless forloop.last -%}<div>and </div>{%- endunless -%}
+{%- else -%}
+{% if forloop.first == true %}
+{%- unless forloop.last -%}<div>, </div>{%- endunless -%}
+{%- else -%}
+{%- unless forloop.last -%}<div>and </div>{%- endunless -%}
 {%- endif -%}
+{%- endif -%}
+{% endfor %}
+{%- endif -%}
```

Comment:
@nykka Please use indentation in all these conditionals. Otherwise it's hard to read